### DISCUSSION TITLE
Alt text vs. longdesc

### DISCUSSION THREAD

**Discussion by BitwiseAndrea** [w3c/wcag#2555] [Q&A] [answered]
Related to the issue here: https://github.com/w3c/wcag/issues/2126
And the awesome discussion linking to the issue here: https://github.com/elastic/eui/pull/5969

For my employers implementation of allowing users to provide their own alt text we're currently limiting them to 100 characters. We can increase this fairly easily, but based on general research decided to stick with 100 as a start.

I would be interested in any thoughts around programmatically deciding based on length, whether or not to use alt text or alt text + longdesc. A lot of the images users will be uploading contain images of text and in the documentation for this feature I'm finding it hard to write what i consider a good alt text for a sample image within 100 characters.

**Comment by JAWS-test** [comment, marked as answer]
There is no default how long an alt text can be. But the limitation to 100 characters makes sense. The background is that e.g. with the screen reader it is not possible to navigate through the alt text as it is possible with other texts. I.e. I always have to listen to the entire alt text from beginning to end and cannot, for example, repeat individual sections. If the alternative text contains structured content (tables, lists, headings), it is even more important that it is not in the alt attribute, because the alt attribute does not allow structured content. There are recommendations for long alternative text on the following pages:
* https://www.w3.org/WAI/tutorials/images/complex/
* https://www.w3.org/WAI/WCAG21/Techniques/general/G95.html

Please note, however, that [longdesc is hardly supported by browsers](https://codepen.io/fstrr/full/jOZZQMP) and should therefore be avoided.
